test/e2e/_helpers.js

```diff
@@ -5,21 +5,46 @@ import { join } from 'path'
 import { format } from 'date-fns'
 import faker from 'faker'
 import glob from 'glob'
-import { find, isArray, isNil } from 'lodash'
-import { ClientFunction, t } from 'testcafe'
+import { find, get, isArray, isNil } from 'lodash'
+import { ClientFunction, RequestLogger, t } from 'testcafe'
 
+import moveService from '../../common/services/move'
 import personService from '../../common/services/person'
+import referenceDataService from '../../common/services/reference-data'
+import { formatDate } from '../../config/nunjucks/filters'
+
+let referenceData
+let genders
+let ethnicities
+let locations
+
+const getReferenceData = async () => {
+  genders = await referenceDataService.getGenders()
+  genders = genders.filter(gender => gender.title.match(/^(Male|Female)$/i))
+  ethnicities = await referenceDataService.getEthnicities()
+  locations = await referenceDataService.getLocations()
+  referenceData = true
```

Comment:
This feels a little flakey. The move fixture uses `locations`, but it relies on the the fact that a person has been created at some point in time for set this value to `true`.

I think it would be safer to have `getReferenceData` check if there is a cached version locally rather than doing that in the fixture. If it exists, return it, if not, make a call to the relevant service and save that value.

There will be a need to create moves without a "person" for the allocations work so it would be good to know that this won't cause us issues when we start testing that.

Something like this:

```javascript
const cachedReferenceData = {}

const getReferenceData = async (key) => {
  if (Object.keys(cachedReferenceData).length) {
    return cachedReferenceData[key]
  }

  cachedReferenceData = {
    genders: await referenceDataService.getGenders().then(genders => genders.filter(gender => gender.title.match(/^(Male|Female)$/i))),
    ethnicities: await referenceDataService.getEthnicities(),
    locations: await referenceDataService.getLocations(),
  }
}


// to get some reference data
getReferenceData('locations')
```

Reply:
Updated to call the ref data service directly and rely on the api-client's cacheing

Reply:
Yep, much nicer and neater 👍